compiler/src/iree/compiler/Codegen/Common/IREEComprehensiveBufferizePass.cpp

```diff
@@ -57,13 +83,15 @@ class EliminateEmptyTensorsPass
 };
 
 /// Pass to convert from tensor based ops to memref based ops.
-class IREEComprehensiveBufferizePass
-    : public IREEComprehensiveBufferizeBase<IREEComprehensiveBufferizePass> {
+class IREEComprehensiveBufferizePass final
+    : public impl::IREEComprehensiveBufferizePassBase<
+          IREEComprehensiveBufferizePass> {
 public:
+  using impl::IREEComprehensiveBufferizePassBase<
+      IREEComprehensiveBufferizePass>::IREEComprehensiveBufferizePassBase;
   explicit IREEComprehensiveBufferizePass(
-      std::optional<BufferizationOptions::AllocationFn> allocationFn =
-          std::nullopt,
-      std::optional<BufferizationOptions::MemCpyFn> memCpyFn = std::nullopt)
+      BufferizationOptions::AllocationFn allocationFn,
```

Comment:
Can we do 
```
IREEComprehensiveBufferizePass(
    BufferizationOptions::AllocationFn allocationFn = defaultAllocationFn,
    BufferizationOptions::AllocationFn memCpyFn = defaultMemCpyFn) : 
 allocationFn(allocationFn), memCpyFn(memCpyFn) {}
```

Reply:
Discussed it offline, it is not feasible because it conflicts with the constructor generated by tablegen.